conf/test_route_control.py

```diff
@@ -0,0 +1,451 @@
+                ("RTA_OIF", 2),
+                ("RTA_DST", "192.168.1.0"),
+            ],
+            "header": {
+                "length": 68,
+                "type": 24,
+                "target": "localhost",
+                "stats": {"qsize": 0, "delta": 0, "delay": 0},
+            },
+            "event": "RTM_NEWROUTE",
+        }
+        result = self.route_controller._parse_route_entry_msg(
+            example_route_entry
+        )
+        self.assertIsInstance(result, RouteEntry)
+        self.assertEqual(result.dest_prefix, "192.168.1.0")
+        self.assertEqual(result.next_hop_ip, "172.31.48.1")
+        self.assertEqual(result.interface, self.ipdb.interfaces[2].ifname)
+        self.assertEqual(result.prefix_len, 24)
+
+    def test_given_valid_route_message_and_dst_len_is_zero_when_parse_message_then_parses_message_as_default_route(
+        self,
+    ):
+        self.ipdb.interfaces = {2: Mock(ifname='core')}
+        example_route_entry = {
+            "family": 2,
+            "dst_len": 0,
+            "flags": 0,
+            "attrs": [
+                ("RTA_TABLE", 254),
+                ("RTA_PRIORITY", 100),
+                ("RTA_PREFSRC", "172.31.55.52"),
+                ("RTA_GATEWAY", "172.31.48.1"),
+                ("RTA_OIF", 2),
+            ],
+            "header": {
+                "length": 68,
+                "type": 24,
+                "target": "localhost",
+                "stats": {"qsize": 0, "delta": 0, "delay": 0},
+            },
+            "event": "RTM_NEWROUTE",
+        }
+        result = self.route_controller._parse_route_entry_msg(
+            example_route_entry
+        )
+        self.assertIsInstance(result, RouteEntry)
+        self.assertEqual(result.dest_prefix, "0.0.0.0")
+        self.assertEqual(result.next_hop_ip, "172.31.48.1")
+        self.assertEqual(
+            result.interface, self.ipdb.interfaces[2].ifname
+        )
+        self.assertEqual(result.prefix_len, 0)
+
+    def test_given_invalid_route_message_when_parse_message_then_returns_none(
+        self
+    ):
+        self.ipdb.interfaces = {2: Mock(ifname='not the needed interface')}
+        example_route_entry = {
+            "family": 2,
+            "flags": 0,
+            "attrs": [
+                ("RTA_TABLE", 254),
+                ("RTA_PRIORITY", 100),
+                ("RTA_PREFSRC", "172.31.55.52"),
+                ("RTA_GATEWAY", "172.31.48.1"),
+                ("RTA_OIF", 2),
+            ],
+            "header": {
+                "length": 68,
+                "type": 24,
+                "target": "localhost",
+                "stats": {"qsize": 0, "delta": 0, "delay": 0},
+            },
+            "event": "RTM_NEWROUTE",
+        }
+        result = self.route_controller._parse_route_entry_msg(
+            example_route_entry
+        )
+        self.assertIsNone(result)
+
+    @patch.object(RouteController, "_handle_new_route_entry")
```

Comment:
We should not patch private methods. With the state of the tests right now (we patch a `_parse_route_entry_msg`, `_parse_route_entry_msg`, `_probe_addr` and more), we can't refactor the code and expect the unit tests to still pass.